test/Dialect/FIRRTL/specialize-option.mlir

```diff
@@ -0,0 +1,29 @@
+// RUN: circt-opt --pass-pipeline='builtin.module(firrtl.circuit(firrtl-specialize-option{select=Platform=FPGA}))' %s | FileCheck %s
```

Comment:
Checking that the specialization of two options works would be good. This can be done with a second `RUN:` line and re-use the same test.